file-attribute-caching/src/test/kotlin/com/pkware/file/fileAttributeCaching/FileAttributeCachingFilesystemTests.kt

```diff
@@ -0,0 +1,338 @@
+            Files.newOutputStream(cachingPath).use { outputStream ->
+                outputStream.write("hello".toByteArray(Charsets.UTF_8))
+            }
+            val lookupService = it.userPrincipalLookupService
+            val owner = lookupService.lookupPrincipalByName("testUser")
+            val group = lookupService.lookupPrincipalByGroupName("testGroup")
+            val permissions = EnumSet.of(
+                PosixFilePermission.OWNER_EXECUTE,
+                PosixFilePermission.OWNER_READ,
+                PosixFilePermission.OWNER_WRITE,
+                PosixFilePermission.GROUP_READ,
+                PosixFilePermission.OTHERS_WRITE,
+            )
+
+            Files.setAttribute(cachingPath, "posix:owner", owner)
+            Files.setAttribute(cachingPath, "posix:group", group)
+            Files.setAttribute(cachingPath, "posix:permissions", permissions)
+            val attributesMap = Files.readAttributes(cachingPath, "posix:*")
+
+            assertThat(attributesMap.size).isEqualTo(12)
+            val ownerUserPrincipal = attributesMap["posix:owner"] as UserPrincipal
+            assertThat(ownerUserPrincipal.name).isEqualTo(owner.name)
+            val groupUserPrincipal = attributesMap["posix:group"] as GroupPrincipal
+            assertThat(groupUserPrincipal.name).isEqualTo(group.name)
+            @Suppress("UNCHECKED_CAST")
+            assertThat(
+                PosixFilePermissions.toString(attributesMap["posix:permissions"] as? MutableSet<PosixFilePermission>)
+            ).isEqualTo(
+                PosixFilePermissions.toString(permissions)
+            )
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(
+        strings = [
+            "dos:readonly",
+            "dos:hidden",
+            "dos:archive",
+            "dos:archive",
+        ]
+    )
+    fun `set and read dos boolean attributes for path`(
+        attributeName: String,
+    ) = windowsJimfs().use { jimfs ->
+        FileAttributeCachingFileSystem.wrapping(jimfs).use {
+
+            // get file attribute caching path
+            val cachingPath = it.getPath("testfile.txt")
+            Files.createFile(cachingPath)
+            Files.newOutputStream(cachingPath).use { outputStream ->
+                outputStream.write("hello".toByteArray(Charsets.UTF_8))
+            }
+
+            Files.setAttribute(cachingPath, attributeName, true)
+            val attributesMap = Files.readAttributes(cachingPath, attributeName)
+
+            assertThat(attributesMap.size).isEqualTo(1)
+            assertThat(attributesMap[attributeName]).isEqualTo(true)
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("allFileSystemsWithCopyOption")
+    fun `copy file from source to target`(
+        option: CopyOption,
+        fileSystem: () -> FileSystem
+    ) = fileSystem().use { jimfs ->
+        FileAttributeCachingFileSystem.wrapping(jimfs).use {
+            // get file attribute caching path
```

Comment:
```suggestion
```

Reply:
not sure why i need to get rid of this comment here when the other tests have it and thats still occuring

Reply:
I didn't leave the comment everywhere, but it's strange to me that the a trivial part of the test has a comment and nothing else does.